tests/test_share.py

```diff
@@ -196,3 +198,10 @@ def test_title_updates_with_query(self, mock_search):
         })
         title = response.xml.find('{http://www.w3.org/2005/Atom}title')
         assert_equal(title.text, 'SHARE: Atom Feed for query: "cats"')
+
+    def test_illegal_unicode_sub(self):
+        illegal_str = u'\u0000\u0008\u000b\u000c\u000e\u001f\ufffe\uffffHello'
+        illegal_str += unichr(0xd800) + unichr(0xdbff) + ' World'
+        assert_equal(util.illegal_unicode_replace(illegal_str), 'Hello World')
+        assert_equal(util.illegal_unicode_replace(''), '')
+        assert_equal(util.illegal_unicode_replace('WOOOooooOOo'), 'WOOOooooOOo')
```

Comment:
Perhaps test for `None` input as well; I believe this was what was breaking in one of your previous commits.